index.js

```diff
@@ -17,17 +17,33 @@ app.post('/', upload.single('thumb'), function (req, res, next) {
         headers: {'content-type' : 'application/x-www-form-urlencoded'},
         url:     'https://slack.com/api/users.profile.set',
         body:    `${common}&profile=%7B%22status_text%22%3A%22${encodeURIComponent(payload.Metadata.originalTitle || payload.Metadata.grandparentTitle)}%20-%20${encodeURIComponent(payload.Metadata.title)}%22%2C%22status_emoji%22%3A%22%3A${process.env.PLAY_EMOJI}%3A%22%7D`
-      }, (error, response, body) => {});
+      }, (error, response, body) => {
+        const parsedBody = !error && response.statusCode === 200 && JSON.parse(body);
+        const errorMessage = error || (!parsedBody.ok && parsedBody.error);
+
+        if (errorMessage) {
+          console.log('error posting play status: ' + errorMessage);
+        }
+      });
     } else if (payload.event == "media.pause" || payload.event == "media.stop") {
       request.post({
         headers: {'content-type' : 'application/x-www-form-urlencoded'},
         url:     'https://slack.com/api/users.profile.set',
         body:    `${common}&profile=%7B%22status_text%22%3A%22%22%2C%22status_emoji%22%3A%22%3A${process.env.PAUSE_EMOJI}%3A%22%7D`
-      }, (error, response, body) => {});
+      }, (error, response, body) => {
+        const parsedBody = !error && response.statusCode === 200 && JSON.parse(body);
+        const errorMessage = error || (!parsedBody.ok && parsedBody.error);
+
+        if (errorMessage) {
+          console.log('error posting stopped / paused status: ' + errorMessage);
+        }
+      });
     }
   }
 
   res.sendStatus(200);
 });
 
-app.listen(process.env.PORT || 10000);
+const port = process.env.PORT || 10000;
+app.listen(port);
```

Comment:
If we're going to wait for the app to be listening, this should be:

```JS
app.listen(port, () => {
  console.log('listening on port ' + port);
});
```

Reply:
I was gonna say, will the log ever fire as written? I thought `listen` was blocking.

Reply:
Blocking! No way! The whole program would fail to run in that case. Gotta keep that event loop free.